File: algorithms_and_data_structure/Two_Number_Sum_3/main.go
package main

import "fmt"

func TwoSum(array []int, target int) []int {
	for i := 0; i < len(array)-1; i++ {
		for j := i + 1; j < len(array); j++ {
			if array[i]+array[j] == target {
				return []int{i, j} // Вернуть индексы, а не сами числа!
			}
		}
	}
	return []int{} // Если не найдено
}

func main() {
	array := []int{3, 4, 5, 6}
	target := 7
	results := TwoSum(array, target)
	fmt.Println(results) // Ожидаемый вывод: [0,1]

	main1()
	main3()
}

func main1() {
	array := []int{4, 5, 6}
	target := 10
	results := TwoSum(array, target)
	fmt.Println(results) // Ожидаемый вывод: [0,2]
}

func main3() {
	array := []int{5, 5}
	target := 10
	results := TwoSum(array, target)
	fmt.Println(results) // Ожидаемый вывод: [0,1]
}

File: algorithms_and_data_structure/Two_Number_Sum_3/main.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            for j in range(i + 1, len(nums)):
                if nums[i] + nums[j] == target:
                    return [i, j]
        return []

# Пример использования:
sol = Solution()

nums = [2, 7, 11, 15]
target = 9
result = sol.twoSum(nums, target)

print("Результат:", result)  # ➜ [0, 1] (потому что 2 + 7 = 9)
